pkg/mkiosdk/mkio_types.go

```diff
@@ -1,7 +1,112 @@
 package mkiosdk
 
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"time"
+)
+
 // Generic types
 
+type MkioClient struct {
+	host             string
+	subscriptionName string
+	token            string
+	hc               *http.Client
+}
+
 type ClientOptions struct {
 	host string
 }
+
+// Do backoff to handle rate limiting
+var backoffSchedule = []time.Duration{
```

Comment:
I believe the default API rate-limiting window is 1 minute - the cumulative backoff durations should ensure we exceed this